app_dart/lib/src/request_handlers/github_webhook.dart

```diff
@@ -87,24 +87,39 @@ class GithubWebhook extends RequestHandler<Body> {
             event,
             existingLabels,
           );
+        } else {
+          await _cancelLuci(
```

Comment:
We should cancel if the PR is closed without merging.  If it's closed with merging, we should let any unfinished jobs finish.

Reply:
Would you mind adding these comments to the source code? For future reviews it will be easier to follow the code along with the comments.